push/ups_integration_with_openshift.md

```diff
@@ -0,0 +1,122 @@
+# Unified Push Server Integration with OpenShift 
+
+## Introduction
+
+This proposal outlines some developer experience issues with provisioning, setting up and configuring the Unified Push Server (UPS) on OpenShift, and integrating it with Apps using the Aerogear SDK.
+
+## Problem Description
+
+Rather than highlighting problems, User Stories will be used to capture improvements and value that can be added that makes for a better overall experience when using UPS on OpenShift
+
+*As a developer, I want UPS running in my OpenShift namespace, ready to go without any additional setup so I can start to use it immediately*
+
+The Unified Push Server is made up of a Java application & a database (MySQL).
+It also has a dependecy on an auth provider (currently Keycloak) for users of the UPS Administration UI or REST API.
+There is a proof of concept UPS Ansible Playbook Bundle (APB) available.
+However, the APB just provisions UPS and MySQL currently.
+It expects an existing Keycloak instance to be running somewhere with the appropriate Realm, Client & Roles already setup.
+This can be improved to eliminate any additional setup or prerequisites required from the developer.
+
+*As a developer, I want a consistent login experience across all Mobile Services so that I can easily login with the same credentials to administer my services*
+
+As detailed in [Sign On Proposal](../auth/developer-single-sign-on-across-mobile-services.md), a goal with all Mobile Services is to have a consistent login experience by logging in using OpenShift credentials.
```

Comment:
should we mention `oauth-proxy` approach ?

Reply:
Yes, I'll update based on recent decisions

Reply:
updated. I added info about the oauth proxy to the auth provider section rather than here in the problems section